controllers/operands/kubevirt.go

```diff
@@ -441,6 +441,10 @@ func getKVConfig(hc *hcov1beta1.HyperConverged) (*kubevirtcorev1.KubeVirtConfigu
 		config.VMStateStorageClass = *hc.Spec.VMStateStorageClass
 	}
 
+	if hc.Spec.VirtualMachineOptions != nil && hc.Spec.VirtualMachineOptions.DisableFreePageReporting {
+		config.VirtualMachineOptions = &kubevirtcorev1.VirtualMachineOptions{DisableFreePageReporting: &kubevirtcorev1.DisableFreePageReporting{}}
+	}
+
```

Comment:
We got a code smell here. The `getKVConfig` is too complex. Please consider to somehow refactor this function. 

Not Blocking!!

Reply:
I have some ideas, do you agree on addressing this with a followup PR?

Reply:
yes. Either that or we'll take it.